### ISSUE TITLE
Support http config

### ISSUE THREAD

**Issue by roidelapluie** [prometheus-community/json_exporter#33]
- Pass basic auth parameters
- Pass TLS client parameters

**Comment by rustycl0ck** [comment]
Any suggestions on how to achieve this? From [here](https://github.com/prometheus-community/json_exporter/pull/30#issuecomment-643772152):
> we should probably use Prometheus client tls config

Does this mean that for TLS config and for auth, we should use the same configuration file syntax used by prometheus? or does this mean that we should something from the `github.com/prometheus/client_golang/prometheus` package or `github.com/prometheus/client_golang/prometheus/promhttp` somehow? or maybe both?

Otherwise I can directly use regular `crypto/tls` and `net/http` to achieve this easily by replacing
```
        client := &http.Client{}
```
with:
```
        tlsConfig := &tls.Config{}
        transport := &http.Transport{TLSClientConfig: tlsConfig}
        client := &http.Client{Transport: transport}
```

Just trying to understand what is the recommended approach for prometheus exporters.

**Comment by roidelapluie** [comment]
HTTPClientConfig from github.com/prometheus/common/config

and then NewRoundTripperFromConfig

**Comment by rustycl0ck** [comment]
This has been fixed in #49
